perma-payments/perma_payments/tests/test_routes.py

```diff
@@ -292,6 +337,202 @@ def test_index_other_methods(client, index):
+    pr_instance = pr.return_value
+    prepped = mocker.patch('perma_payments.views.prep_for_cybersource', autospec=True)
+
+    # request
+    response = client.post(purchase['route'])
+
+    # assertions
+    assert response.status_code == 200
+    fields_to_prep = {
+        'access_key': settings.CS_ACCESS_KEY,
+        'amount': pr_instance.amount,
+        'currency': pr_instance.currency,
+        'locale': pr_instance.locale,
+        'payment_method': pr_instance.payment_method,
+        'profile_id': settings.CS_PROFILE_ID,
+        'reference_number': pr_instance.reference_number,
+        'signed_date_time': pr_instance.get_formatted_datetime(),
+        'transaction_type': pr_instance.transaction_type,
+        'transaction_uuid': pr_instance.transaction_uuid,
+    }
+    prepped.assert_called_once_with(fields_to_prep)
+    for field in FIELDS_REQUIRED_FOR_CYBERSOURCE['purchase']:
+        assert field in fields_to_prep
+
+
+def test_purchase_post_redirect_form_populated_correctly(client, purchase, purchase_redirect_fields, mocker):
+    # mocks
+    mocker.patch('perma_payments.views.process_perma_transmission', autospec=True, return_value=purchase['valid_data'])
+    mocker.patch('perma_payments.views.transaction.atomic', autospec=True)
+    mocker.patch('perma_payments.views.PurchaseRequest', autospec=True)
+    mocker.patch('perma_payments.views.prep_for_cybersource', autospec=True, return_value=purchase_redirect_fields)
+
+    # request
+    response = client.post(purchase['route'])
+
+    # assertions
+    assert response.status_code == 200
+    assert response.context['fields_to_post'] == purchase_redirect_fields
+    context = list(response.context.keys())
+    for field in ['fields_to_post', 'post_to_url']:
+        assert field in context
+    expected_template_used(response, 'redirect.html')
+    for field in purchase_redirect_fields:
+        assert bytes('<input type="hidden" name="{0}" value="{0}">'.format(field), 'utf-8') in response.content
+
+
+def test_purchase_other_methods(client, purchase):
+    get_not_allowed(client, purchase['route'])
+    put_patch_delete_not_allowed(client, purchase['route'])
+
+
+# acknowledge-purchase
+
+@pytest.mark.django_db
+def test_acknowledge_purchase(client, acknowledge_purchase, purchase_request_response_factory, mocker):
+    mocker.patch('perma_payments.views.process_perma_transmission', autospec=True, return_value=acknowledge_purchase['valid_data'])
+    prr = purchase_request_response_factory(
+        id=SENTINEL['purchase_pk'],
+        inform_perma=True
+    )
+    assert not prr.perma_acknowleged_at
+
+    # request
+    response = client.post(acknowledge_purchase['route'])
+
+    # assertions
+    assert response.status_code == 200
+    prr.refresh_from_db()
+    assert prr.perma_acknowleged_at
+
+
+@pytest.mark.django_db
+def test_acknowledge_unacknowledgable_purchase(client, acknowledge_purchase, purchase_request_response_factory, mocker):
```

Comment:
I think I'd spell this `test_acknowledge_unacknowledgeable_purchase` but it certainly doesn't matter.